### PR TITLE
Set repository version based on target branch

### PR DESCRIPTION
**Pull request by pablochacin**
## Why is this PR needed?

For e2e tests and pr tests that specify a repository branch, set the version of the repository based on the target branch
(master or maintenance).

Fixes https://github.com/SUSE/avant-garde/issues/1985

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

File: ci/jenkins/pipelines/prs/skuba-test.Jenkinsfile [hunk 1]
```diff
@@ -89,10 +90,14 @@ node('caasp-team-private-integration') {
            }
            if (pr_repo_label != null) {
                def branch_name = pr_repo_label.name.split(":")[1]
-               branch_repo = "http://download.suse.de/ibs/Devel:/CaaSP:/${repo_version}:/Branches:/${branch_name}/SLE_15_SP2"
-               branch_registry = "registry.suse.de/devel/caasp/4.5/branches/${branch_name}/containers"
-               original_registry = "registry.suse.de/devel/caasp/4.5/containers/containers"
-           }
+               if (env.CHANGE_TARGET.startsWith('maintenance')){
```
Comment (dannysauer) [inline ci/jenkins/pipelines/prs/skuba-test.Jenkinsfile:93]: Should this also check if it includes "4.2" in the string?  At some point, I assume 4.5 will also be maintenance, as we move to integrating with Rancher.

I'm not sure what happens to skuba at that point, and no one else is either. :)  But this is sort of a hidden thing I'm not sure any of us will remember to update if that happens, so I don't know if it makes more sense to prepare for that possibility by making the code slightly more complex, or if there should be a document somewhere recording "things we will have to remember to change if 4.5 has a maintenance branch", or something else.  Or maybe this concern is all just premature optimization. :D

**Review by dannysauer — APPROVED** [review]
My question about the code shouldn't prevent merging.  I think this looks like a solid change.

File: ci/jenkins/pipelines/prs/skuba-test.Jenkinsfile (hunk 1 above)
Comment (pablochacin) [reply ci/jenkins/pipelines/prs/skuba-test.Jenkinsfile:93]: It's a good point, but this is an unlikely scenario. We can manage it if this ever happens

**Review by chentex — APPROVED** [review]
LGTM

**Review by davidcassany — APPROVED** [review]
LGTM

**Review by chentex — APPROVED** [review]

**Pull request merged**